Sources/SpeziScheduler/SchedulerStorage.swift

```diff
@@ -17,32 +17,45 @@ protocol AnyStorage {
 }
 
 
-actor SchedulerStorage<Context: Codable>: Module, DefaultInitializable, AnyStorage {
+@_spi(Spezi)
+/// Underlying Storage module for the `Scheduler`.
+public actor SchedulerStorage<Context: Codable>: Module, DefaultInitializable, AnyStorage {
     private let logger = Logger(subsystem: "edu.stanford.spezi.scheduler", category: "Storage")
 
     @Dependency private var localStorage: LocalStorage
 
-    private let taskList: TaskList<Context>
+    private let storageIsMocked: Bool
+    let taskList: TaskList<Context>
 
     private var debounceTask: _Concurrency.Task<Void, Never>? {
         willSet {
             debounceTask?.cancel()
         }
     }
+    
 
-
-    init() {
-        self.taskList = TaskList()
+    public init() {
+        #if targetEnvironment(simulator)
+        self.init(mockedStorage: true)
+        #else
+        self.init(mockedStorage: false)
+        #endif
     }
-    
 
-    init(taskList: TaskList<Context>) {
-        self.taskList = taskList
+    public init(for scheduler: Scheduler<Context>.Type = Scheduler<Context>.self, mockedStorage: Bool) {
+        // swiftlint:disable:previous function_default_parameter_at_end
+        self.taskList = TaskList()
+        self.storageIsMocked = mockedStorage
     }
 
 
     func loadTasks() -> [Task<Context>]? {
         // swiftlint:disable:previous discouraged_optional_collection
+        if storageIsMocked {
+            logger.info("Storage is disabled as we are running int the simulator. No tasks were loaded.")
```

Comment:
Thank you for the additional log messages! It would be great to add a hint to a mechanism to test this locally with persistence if needed within the logger message. Can be a very short code example.

Reply:
Added a multiline log message illustrating usage of the initializer.